Resources/gametree.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+from collections import deque
+from dataclasses import dataclass, field
+from typing import Any
+
+"""
+The purpose of this class is to serve as data storage for
+State exploration. Comparison operators can be used to determine
+reward. The dataclass feature requires python 3.7+.
+"""
+
+@dataclass(order=True)
+class node:
+    reward:         int
+    state:          Any=field(compare=False)
+    attackLeft:     Any=field(compare=False)
+    forward:        Any=field(compare=False)
+    attackRight:    Any=field(compare=False)
+
+    def __init__(self, reward, state):
+        self.reward = reward
+        self.state = None
```

Comment:
Why do we take a state arg if we don't use it?

Reply:
I noticed this too, it's fixed on my local -  I just haven't pushed it yet. I'll push new updates before merging.